config/v1/types_oauth.go

```diff
@@ -204,111 +184,92 @@ type IdentityProviderConfig struct {
-	// connecting to the server.
-	// Looks under the key "tls.cert" for the data unless a lookup key is specified in the secret ref
-	TLSClientCert LocalSecretReference `json:"tlsClientCert"`
-
-	// tlsClientKey references a secret containing the TLS private key for the client certificate
-	// Looks under the key "tls.key" for the data unless a lookup key is specified in the secret ref
-	TLSClientKey LocalSecretReference `json:"tlsClientKey"`
+	// ca is a reference to a config map by name containing the CA for verifying TLS connections.
+	// The key "ca.crt" is used to locate the data.
+	CA string `json:"ca"`
+
+	// tlsClientCert references a secret by name that contains the PEM-encoded
+	// TLS client certificate to present when connecting to the server.
+	// The key "tls.crt" is used to locate the data.
+	TLSClientCert string `json:"tlsClientCert"`
+
+	// tlsClientKey references a secret by name that contains the PEM-encoded
+	// TLS private key for the client certificate referenced in tlsClientCert.
+	// The key "tls.key" is used to locate the data.
+	TLSClientKey string `json:"tlsClientKey"`
 }
 
-// HTPasswdDataKey is the default key for the htpasswd file data in a secret
-const HTPasswdDataKey = "htpasswd"
-
 // HTPasswdPasswordIdentityProvider provides identities for users authenticating using htpasswd credentials
-type HTPasswdPasswordIdentityProvider struct {
-	// fileData is a reference to a secret containing the data to use as the htpasswd file
-	// Looks under the key `htpasswd` unless a lookup key is specified in the secret ref
-	FileData LocalSecretReference `json:"fileData"`
+type HTPasswdIdentityProvider struct {
+	// fileData is a reference to a secret by name containing the data to use as the htpasswd file.
+	// The key "htpasswd" is used to locate the data.
+	FileData string `json:"fileData"`
 }
 
-const (
-	// BindPasswordKey is default the key for the LDAP bind password in a secret
-	BindPasswordKey = "bindPassword"
-	// ClientSecretKey is the key for the oauth client secret data in a secret
-	ClientSecretKey = "clientSecret"
-)
-
 // LDAPPasswordIdentityProvider provides identities for users authenticating using LDAP credentials
-type LDAPPasswordIdentityProvider struct {
+type LDAPIdentityProvider struct {
 	// url is an RFC 2255 URL which specifies the LDAP search parameters to use.
 	// The syntax of the URL is:
-	//    ldap://host:port/basedn?attribute?scope?filter
+	// ldap://host:port/basedn?attribute?scope?filter
 	URL string `json:"url"`
 
 	// bindDN is an optional DN to bind with during the search phase.
 	// +optional
 	BindDN string `json:"bindDN"`
 
-	// bindPassword is a reference to the secret containing an optional password to bind
+	// bindPassword is a reference to a secret by name containing an optional password to bind
 	// with during the search phase.
-	// Looks under the key `bindPassword` unless a lookup key is specified in the secret ref
+	// The key "bindPassword" is used to locate the data.
```

Comment:
Are the keys case sensitive (if so, wondering if it should be camelCase)

Reply:
It is a standard go map so it would be case sensitive.